### PR TITLE
Add default prom metrics to scaffolder

### PR DESCRIPTION
**Pull request by spencerrichardhenry**
## Hey, I just made a Pull Request!

This is for adding a set of default scaffolder metrics to Backstage.
I managed to botch the git history enough on my previous PR that it seemed simpler to just re-open here and squash the changes into one commit. Old PR: https://github.com/backstage/backstage/pull/13683

github issue:
https://github.com/backstage/backstage/issues/12453
#### :heavy_check_mark: Checklist

- [:heavy_check_mark:] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [:heavy_check_mark:] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder-backend/package.json
Comment (freben) [inline plugins/scaffolder-backend/package.json:75]: should have a `^` on it (and re-run yarn after)

**Review by freben — COMMENTED** [review]
Alright! Almost there

**Review by awanlin — APPROVED** [review]
Looks good to me, pretty excited to see this in place, thanks again for all the work on this! 🚀

**Review by angeliski — APPROVED** [review]
Amazing work! Could I suggest to update https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/prometheus-metrics.md#experimental to list those new metrics?

**Comment by spencerrichardhenry** [comment]
great suggestion thanks @angeliski, added the metrics to that tutorial doc

**Comment by spencerrichardhenry** [comment]
Would love some guidance on these E2E test failures. No clue what they mean or how they'd be connected to my changes. TIA

**Comment by Rugvip** [comment]
@spencerrichardhenry I'd say start off by rebasing on latest master, it might just be stuck with a broken octokit version or something like that. If that doesn't fix it I can help dig a bit more

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 

Couple of tweaks following discussion with the rest of the maintainers

File: .changeset/flat-items-perform.md
Comment (spencerrichardhenry) [reply .changeset/flat-items-perform.md:26]: great catch

File: plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts
Comment (spencerrichardhenry) [reply plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts:358]: was OOO I'll get this changed today!

File: plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts
Comment (Rugvip) [inline plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts:339]: Just to keep things consistent I'm thinking it's best to just keep all of these async tbh, feeling that would reduce the number of surprises in the code.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice, thank you! 👍 

Couple more tweaks and I'm feeling we can ship it. Think we'll target to merge this after the main-line release next Tuesday, as it's a bit too risky to ship this straight into a main release

File: plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts
Comment (spencerrichardhenry) [reply plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts:339]: Sure makes sense

File: plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts
Comment (spencerrichardhenry) [reply plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts:233]: Great catch this seems to have gotten shuffled in all my rebasing messs

File: plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts
Comment (spencerrichardhenry) [reply plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts:358]: 🤦 thanks

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice! 👍

Found a minor thing from before that we might as well fix, and then there are some bonus commits we need to drop.

File: packages/cli/src/lib/bundler/config.ts
Comment (Rugvip) [reply packages/cli/src/lib/bundler/config.ts]: There are a couple of ways to get rid of them, but for this kind of thing I'd usually go with a `git rebase --interactive` and then delete the lines for the comments you want to remove.

Comment (spencerrichardhenry) [reply packages/cli/src/lib/bundler/config.ts]: OK I believe that's resolved now

Comment (spencerrichardhenry) [reply packages/cli/src/lib/bundler/config.ts]: but now there are merge conflicts because those commits are missing. I don't understand.

Comment (spencerrichardhenry) [reply packages/cli/src/lib/bundler/config.ts]: OK I resolved the merge conflict and it looks like things will be cleared up now

**Comment by spencerrichardhenry** [comment]
@Rugvip I think this is ready to merge now, there are 'requested changes' that you might need to close out still?

**Review by Rugvip — APPROVED** [review]
I was gonna go get rid of the unnecessary commits, but w/e, let's :shipit: 😁

**Comment by awanlin** [comment]
Awesome work @spencerrichardhenry, many people are going to be happy with this!

**Comment by suuus** [comment]
Thank you for contributing during Hacktoberfest 🍂! You can claim the Backstage Hacktoberfest Holopin at https://bck.st/hacktoberfest-holopin 🙌🏻

**Pull request merged**
